ouroboros-consensus-cardano/app/db-converter.hs

```diff
@@ -0,0 +1,266 @@
+            , (doesFileExist, ["tables", "tvar"], "Input mode is Mem but the provided path DOES NOT contain a \"tables/tvar\" file")
+            ]
+    LMDB ->
+        mapM_
+            (\(doCheck, extra :: [Text], err) -> want (doCheck fs) (foldl' (</>) p extra) err)
+            [ (doesDirectoryExist, [], "Input mode is LMDB but the provided path is NOT a directory")
+            , (doesFileExist, ["state"], "Input mode is LMDB but the provided path DOES NOT contain a \"state\" file")
+            , (doesDirectoryExist, ["tables"], "Input mode is LMDB but the provided path DOES NOT contain a \"tables\" directory")
+            , (doesFileExist, ["tables", "data.mdb"], "Input mode is LMDB but the provided path DOES NOT contain a \"tables/data.mdb\" file")
+            ]
+  where
+    want :: (FsPath -> IO Bool) -> FsPath -> String -> ExceptT Error IO ()
+    want fileType path err = ExceptT $ do
+        exists <- fileType path
+        pure $ if exists then Right () else Left $ UnsuitableSnapshot err
+
+load ::
+       forall blk.
+       ( LedgerDbSerialiseConstraints blk
+       , CanStowLedgerTables (LedgerState blk)
+       , HasLedgerTables (LedgerState blk)
+       )
+    => Config
+    -> SomeHasFS IO
+    -> CodecConfig blk
+    -> ExceptT Error IO (ExtLedgerState blk ValuesMK)
+load Config{from = Legacy, inpath} fs ccfg = do
+    checkSnapshot Legacy inpath fs
+    withExceptT SnapshotError $
+        unstowLedgerTables <$> readExtLedgerState fs (decodeDiskExtLedgerState ccfg) decode inpath
+load Config{from = Mem, inpath} fs@(SomeHasFS hasFS) ccfg = do
+    checkSnapshot Mem inpath fs
+    extLedgerSt <- withExceptT SnapshotError $ readExtLedgerState fs (decodeDiskExtLedgerState ccfg) decode (inpath </> "state")
+    values <- ExceptT $ withFile hasFS (inpath </> "tables" </> "tvar") ReadMode $ \h -> do
+        bs <- hGetAll hasFS h
+        case CBOR.deserialiseFromBytes valuesMKDecoder bs of
+            Left err -> pure $ Left $ TablesCantDeserializeError err
+            Right (extra, x) ->
+                pure $
+                    if BSL.null extra
+                        then Right x
+                        else Left TablesTrailingBytes
+    pure (extLedgerSt `withLedgerTables` values)
+load Config{from = LMDB, inpath} fs ccfg = do
+    checkSnapshot LMDB inpath fs
+    extLedgerSt <- withExceptT SnapshotError $ readExtLedgerState fs (decodeDiskExtLedgerState ccfg) decode (inpath </> "state")
+    values <- liftIO $ do
+        dbEnv <- LMDB.openEnvironment (fsToFilePath (MountPoint ".") (inpath </> "tables")) defaultLMDBLimits
+        Disk.LMDBMK _ dbBackingTables <- LMDB.readWriteTransaction dbEnv (Disk.getDb (K2 "utxo"))
+        LMDB.readOnlyTransaction dbEnv $
+            LMDB.Cursor.runCursorAsTransaction'
+                LMDB.Cursor.cgetAll
+                dbBackingTables
+                (fromCodecMK $ getLedgerTables $ codecLedgerTables @(LedgerState blk))
```

Comment:
`TablesCantDeserialiseError` should probably be thrown here too in case the key/value types are wrong